oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/web/DefaultBearerTokenResolver.java

```diff
@@ -36,40 +39,42 @@
  */
 public final class DefaultBearerTokenResolver implements BearerTokenResolver {
 
-	private static final Pattern authorizationPattern = Pattern.compile("^Bearer (?<token>[a-zA-Z0-9-._~+/]+)=*$");
+	protected static final Pattern AUTHORIZATION_PATTERN = Pattern.compile("^Bearer (?<token>[a-zA-Z0-9-._~+/]+)=*$");
 
-	private boolean allowFormEncodedBodyParameter = false;
+	protected static final String ACCESS_TOKEN_PARAM_NAME = "access_token";
 
-	private boolean allowUriQueryParameter = false;
+	protected boolean allowFormEncodedBodyParameter = false;
+
+	protected boolean allowUriQueryParameter = false;
 
 	/**
 	 * {@inheritDoc}
 	 */
 	@Override
-	public String resolve(HttpServletRequest request) {
-		String authorizationHeaderToken = resolveFromAuthorizationHeader(request);
-		String parameterToken = resolveFromRequestParameters(request);
-		if (authorizationHeaderToken != null) {
-			if (parameterToken != null) {
-				BearerTokenError error = new BearerTokenError(BearerTokenErrorCodes.INVALID_REQUEST,
-						HttpStatus.BAD_REQUEST,
-						"Found multiple bearer tokens in the request",
-						"https://tools.ietf.org/html/rfc6750#section-3.1");
-				throw new OAuth2AuthenticationException(error);
-			}
-			return authorizationHeaderToken;
+	public String resolve(final HttpServletRequest request) {
+		final String authorizationHeaderToken = resolveFromAuthorizationHeader(request);
+		if (hasMultipleAccessTokens(request, authorizationHeaderToken)) {
+			final BearerTokenError error = new BearerTokenError(BearerTokenErrorCodes.INVALID_REQUEST,
+					HttpStatus.BAD_REQUEST,
+					"Found multiple bearer tokens in the request",
+					"https://tools.ietf.org/html/rfc6750#section-3.1");
+			throw new OAuth2AuthenticationException(error);
 		}
-		else if (parameterToken != null && isParameterTokenSupportedForRequest(request)) {
-			return parameterToken;
-		}
-		return null;
+		return authorizationHeaderToken != null ? authorizationHeaderToken : resolveFromRequestParameters(request);
+	}
+
+	protected boolean hasMultipleAccessTokens(final HttpServletRequest request, final String authorizationHeaderToken) {
```

Comment:
Also, if you don't mind, it would be nice if all the public methods went first, then the private methods, see the [style guide](https://github.com/spring-projects/spring-framework/wiki/Code-Style#java-source-file-organization).

Reply:
Will make this change